### PR TITLE
feat(output): Generate dot file and support multi outformat.(#441) Re

### PR DESCRIPTION
**Pull request by gabriel-tessier**
I lost this branch on my local but I was able to pull the pr, but I can no more link it with the previous PR so I open a new one.
Sorry for the inconvenience.

Replace the previous PR https://github.com/mingrammer/diagrams/pull/446

### CODE REVIEW

**Comment by danielkocot** [comment]
To integrate diagrams in existing docs-as-code toolchains this feature is absolutely needed. Thanks @gabriel-tessier! Any ideas when the pr will be merged and a new release will follow?

**Comment by forcewake** [comment]
Thanks @gabriel-tessier for this PR

@mingrammer I saw that you approved previous PR with changes for dot file generation - do you have any plans to approve and merge this PR?
Thank you!

File: diagrams/__init__.py [hunk 1]
```diff
@@ -260,9 +266,8 @@ def __exit__(self, exc_type, exc_value, traceback):
 
     def _validate_direction(self, direction: str):
         direction = direction.upper()
-        for v in self.__directions:
-            if v == direction:
-                return True
+        if direction in self.__directions:
+            return True
         return False
```
Comment (mingrammer) [inline diagrams/__init__.py:271]: We can do it by just `return direction in self.__directions`.

File: diagrams/__init__.py [hunk 2]
```diff
@@ -156,23 +161,20 @@ def _repr_png_(self):
...
 
     def _validate_curvestyle(self, curvestyle: str) -> bool:
         curvestyle = curvestyle.lower()
-        for v in self.__curvestyles:
-            if v == curvestyle:
-                return True
+        if curvestyle in self.__curvestyles:
+            return True
         return False
```
Comment (mingrammer) [inline diagrams/__init__.py:172]: +1

File: diagrams/__init__.py [hunk 3]
```diff
@@ -156,23 +161,20 @@ def _repr_png_(self):
...
 
     def _validate_outformat(self, outformat: str) -> bool:
         outformat = outformat.lower()
-        for v in self.__outformats:
-            if v == outformat:
-                return True
+        if outformat in self.__outformats:
+            return True
         return False
```
Comment (mingrammer) [inline diagrams/__init__.py:178]: +1

File: diagrams/__init__.py [hunk 4]
```diff
@@ -156,23 +161,20 @@ def _repr_png_(self):
 
     def _validate_direction(self, direction: str) -> bool:
         direction = direction.upper()
-        for v in self.__directions:
-            if v == direction:
-                return True
+        if direction in self.__directions:
+            return True
```
Comment (mingrammer) [inline diagrams/__init__.py:165]: +1

**Review by mingrammer — COMMENTED** [review]
LGTM. Please check some minor issues.

**Review by mingrammer — APPROVED** [review]
LGTM! Thank you :)

**Pull request merged**